plugin/route53/setup.go

```diff
@@ -75,11 +82,23 @@ func setup(c *caddy.Controller, f func(*credentials.Credentials) route53iface.Ro
 				if err != nil {
 					return c.Errf("invalid upstream: %v", err)
 				}
+			case "credentials":
+				args := c.RemainingArgs()
+				if len(args) > 0 {
+					sharedProvider.Profile = args[0]
+				}
+				if len(args) > 1 {
+					sharedProvider.Filename = args[1]
+				}
 			default:
 				return c.Errf("unknown property '%s'", c.Val())
 			}
 		}
 	}
+	providers = append(providers, &credentials.EnvProvider{}, sharedProvider)
+
+	credential := credentials.NewChainCredentials(providers)
+
 	client := f(credential)
```

Comment:
You can inline `credentials.NewChainCredentials(providers)`, `credential` is not used anywhere else it seems.